core/src/env/api.rs

```diff
@@ -64,8 +68,20 @@ pub unsafe fn load(key: Key) -> Option<Vec<u8>> {
 /// own to always encode the expected type.
 pub unsafe fn r#return<T, E>(value: T) -> !
 where
-    T: parity_codec::Encode,
+    T: Encode,
     E: Env,
 {
     E::r#return(&value.encode()[..])
 }
+
+/// Dispatches a Call into the runtime, for invoking other substrate
+/// modules. Dispatched only after successful contract execution.
+///
+/// The encoded Call MUST be decodable by the target substrate runtime.
+/// If decoding fails, then the smart contract execution will fail.
+pub fn dispatch_call<T>(call: <T as EnvTypes>::Call)
```

Comment:
Since this is an interfacing method and `Call` is generally an `enum`, can't we here as well use an `Into<<T as EnvTypes>::Call>` generic input instead?